### PR TITLE
e2e: regenerate namespace name if the name is already taken

### PR DESCRIPTION
**Pull request by cofyc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
/kind flake

**What this PR does / why we need it**:
Currently, the name of testing namespace may conflict with others on creation. To make it robust,  we can regenerate if the name is already taken.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -335,10 +316,7 @@ func CreateTestingNS(baseName string, c clientset.Interface, labels map[string]s
 	// We don't use ObjectMeta.GenerateName feature, as in case of API call
 	// failure we don't know whether the namespace was created and what is its
 	// name.
-	name, err := findAvailableNamespaceName(baseName, c)
-	if err != nil {
-		return nil, err
-	}
+	name := fmt.Sprintf("%v-%v", baseName, RandomSuffix())
```
Comment (neolit123) [inline test/e2e/framework/util.go:319]: the old code actually tries to find an avail. name for 30 seconds with 2 second intervals in a loop, so ~15 retires.
this seems OK.

with this change however, we stop doing that an regenerate on conflict a single time in CreateTestingNS.
in my opinion the old code is more robust.

Comment (cofyc) [reply test/e2e/framework/util.go:319]: the problem of old code is before the actual creation, the name may be taken by other processes.

my fix is to regenerate the name in `wait.PollImmediate(Poll, 30*time.Second, func() (bool, error) {`, try a new name each time.

Comment (neolit123) [reply test/e2e/framework/util.go:319]: right, i missed that the creation is already in a loop.

**Comment by neolit123** [comment]
/priority backlog
/retest
/approve
/hold

@alejandrox1 @oomichi PTAL for LGTM.

File: test/e2e/framework/util.go (hunk 1 above)
Comment (neolit123) [reply test/e2e/framework/util.go:319]: thank you for the explanation.

**Review by oomichi — COMMENTED** [review]
I prefer this new way to operate actual creation and try renaming if conflict happens.
That can make operation and time less.

/lgtm
/hold cancel

**Pull request merged**